File: test/aidbox_sdk/snapshots/python/hl7-fhir-r4-core/CareTeam.py
from typing import Optional, List
from base import Annotation
from base import Period
from base import CodeableConcept
from base import ContactPoint
from base import DomainResource
from base import Reference
from base import Identifier
from base import BackboneElement

class CareTeam_Participant(BackboneElement):
    role: Optional[List[CodeableConcept]] = None
    member: Optional[Reference] = None
    period: Optional[Period] = None
    on_behalf_of: Optional[Reference] = None

class CareTeam(DomainResource):
    category: Optional[List[CodeableConcept]] = None
    managing_organization: Optional[List[Reference]] = None
    encounter: Optional[Reference] = None
    name: Optional[str] = None
    reason_code: Optional[List[CodeableConcept]] = None
    participant: Optional[List[CareTeam_Participant]] = None
    note: Optional[List[Annotation]] = None
    status: Optional[str] = None
    identifier: Optional[List[Identifier]] = None
    telecom: Optional[List[ContactPoint]] = None
    period: Optional[Period] = None
    subject: Optional[Reference] = None
    reason_reference: Optional[List[Reference]] = None

File: test/aidbox_sdk/snapshots/dotnet/hl7-fhir-r4-core/CareTeam.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class CareTeam : DomainResource, IResource
{
    public Base.CodeableConcept[]? Category { get; set; }
    public Base.ResourceReference[]? ManagingOrganization { get; set; }
    public Base.ResourceReference? Encounter { get; set; }
    public string? Name { get; set; }
    public Base.CodeableConcept[]? ReasonCode { get; set; }
    public CareTeam_Participant[]? Participant { get; set; }
    public Base.Annotation[]? Note { get; set; }
    public string? Status { get; set; }
    public Base.Identifier[]? Identifier { get; set; }
    public Base.ContactPoint[]? Telecom { get; set; }
    public Base.Period? Period { get; set; }
    public Base.ResourceReference? Subject { get; set; }
    public Base.ResourceReference[]? ReasonReference { get; set; }

    public class CareTeam_Participant : BackboneElement
    {
        public Base.CodeableConcept[]? Role { get; set; }
        public Base.ResourceReference? Member { get; set; }
        public Base.Period? Period { get; set; }
        public Base.ResourceReference? OnBehalfOf { get; set; }
    }
}